pkg/koordlet/runtimehooks/nri/server.go

```diff
@@ -209,6 +221,20 @@ func (p *NriServer) UpdateContainer(pod *api.PodSandbox, container *api.Containe
 }
 
 func (p *NriServer) onClose() {
-	p.stub.Stop()
-	klog.V(6).Infof("NRI server closes")
+	for {
+		stub, err := stub.New(p, append(opts, stub.WithOnClose(p.onClose))...)
+		if err != nil {
+			klog.Errorf("failed to create plugin stub: %v", err)
+		}
+
+		p.stub = stub
+		err = p.Start()
+		if err != nil {
+			time.Sleep(p.options.ReconnectInterval)
```

Comment:
Do we also need a Reconnection limit to control retry times ?

Reply:
Agree. Perhaps a retry with backoff.